fud2/fud-core/src/run.rs

```diff
@@ -297,20 +319,28 @@ impl<'a> Run<'a> {
 
         // Emit the build commands for each step in the plan.
         emitter.comment("build targets")?;
-        let mut last_file = &self.plan.start;
-        for (op, out_file) in &self.plan.steps {
+        for (op, in_files, out_files) in &self.plan.steps {
             let op = &self.driver.ops[*op];
             op.emit.build(
                 &mut emitter,
-                last_file.as_str(),
-                out_file.as_str(),
+                in_files
+                    .iter()
+                    .map(|io| io.filename().as_str())
+                    .collect::<Vec<_>>()
+                    .as_slice(),
+                out_files
+                    .iter()
+                    .map(|io| io.filename().as_str())
+                    .collect::<Vec<_>>()
+                    .as_slice(),
             )?;
-            last_file = out_file;
         }
         writeln!(emitter.out)?;
 
         // Mark the last file as the default target.
```

Comment:
Perhaps this comment needs an update now: "Mark the output files as the default targets."